sklearn/multioutput.py

```diff
@@ -648,6 +648,51 @@ def _log_message(self, *, estimator_idx, n_estimators, processing_msg):
             return None
         return f"({estimator_idx} of {n_estimators}) {processing_msg}"
 
+    def _get_chain_method(self):
```

Comment:
I think that you can simplify a lot this code using `_get_response_values`.

Only the stacking will remain.

Reply:
Actually I think this is needed because `RegressorChain` does not have a `chain_method` parameter.

Reply:
you can have the following in `fit`:

```python
# chain_method is not existing for regressor and predict is used
# as a default
chain_method = getattr(self, "chain_method", "predict")
```

Reply:
Ah yes of course, that is better.

But now that we allow `chain_method` to be a list, I have to use the utility `_check_response_method` to get the preferred method as string. This is because `fit` uses `cross_val_predict` to get the predictions and `cross_val_predict` does not allow its `method` param to be a list.

I also still have the problem where I cannot assign `chain_method_` to self in `_BaseChain.fit` because `RegressorChain` does not require it. Instead I get the chain method  in `_BaseChain.fit` and repeat again `_BaseChain._get_predictions` (which is used in all 'predict' methods). There may be a better solution?

Reply:
I'm silly, I've just assigned `self.chain_method_` in `fit` if `self` had a `chain_method` attr. This should be fine now.